src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java

```diff
@@ -67,22 +80,50 @@ public AbstractMavenArtifactChoiceListProvider(final String artifactId) {
 
     @Override
     public List<String> getChoiceList() {
-
+    	StaplerRequest req = Stapler.getCurrentRequest();
+    	Item item = req.findAncestorObject(Item.class);
```

Comment:
See https://github.com/jenkinsci/stapler/blob/master/core/src/main/java/org/kohsuke/stapler/AncestorInPath.java

Reply:
Hey @yaroslavafenkin,

i think its not possible to use the @AnchestorInPath annotation here, as i have to override/implement get getChoiceList() method from the superclass.

```
@Override
public List<String> getChoiceList() {
    StaplerRequest req = Stapler.getCurrentRequest();
    Item item = req.findAncestorObject(Item.class);
    ....
```

Reply:
Ah, you're right, sorry. Then your current approach works

Reply:
Are we good to build a new version?

Reply:
Let me re-test the fix after the recent changes